src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -299,125 +336,126 @@ void func_80A747C0(EnIk* this, GlobalContext* globalCtx) {
     } else {
-        Animation_Change(&this->skelAnime, &object_ik_Anim_006734, 1.0f, 0.0f,
-                         Animation_GetLastFrame(&object_ik_Anim_006734), ANIMMODE_LOOP, -4.0f);
+        Animation_Change(&this->skelAnime, &gIronKuckleRunAnim, 1.0f, 0.0f, Animation_GetLastFrame(&gIronKuckleRunAnim),
+                         ANIMMODE_LOOP, -4.0f);
         Audio_PlayActorSound2(&this->actor, NA_SE_EN_IRONNACK_DASH);
         this->actor.speedXZ = 2.5f;
     }
     this->actor.world.rot.y = this->actor.shape.rot.y;
-    EnIk_SetupAction(this, func_80A74BA4);
+    EnIk_SetupAction(this, EnIk_WalkOrRun);
 }
 
-void func_80A74BA4(EnIk* this, GlobalContext* globalCtx) {
+void EnIk_WalkOrRun(EnIk* this, GlobalContext* globalCtx) {
     s16 temp_t0;
-    s16 temp_a1;
+    s16 targetStepVal;
     s16 yawDiff;
     s16 sp30;
     s16 sp2E;
-    s16 phi_a3;
+    s16 stepVal;
 
-    if (this->unk_2FB == 0) {
+    if (this->armorStatusFlag == ARMOR_UNBROKEN) {
         temp_t0 = 0xAAA;
-        phi_a3 = 0x320;
+        stepVal = 0x320;
         sp30 = 0;
         sp2E = 0x10;
     } else {
         temp_t0 = 0x3FFC;
-        phi_a3 = 0x4B0;
+        stepVal = 0x4B0;
         sp30 = 2;
         sp2E = 9;
     }
-    temp_a1 = this->actor.wallYaw - this->actor.shape.rot.y;
-    if ((this->actor.bgCheckFlags & BGCHECKFLAG_WALL) && (ABS(temp_a1) >= 0x4000)) {
-        temp_a1 = (this->actor.yawTowardsPlayer > 0) ? this->actor.wallYaw - 0x4000 : this->actor.wallYaw + 0x4000;
-        Math_SmoothStepToS(&this->actor.world.rot.y, temp_a1, 1, phi_a3, 0);
+    targetStepVal = this->actor.wallYaw - this->actor.shape.rot.y;
+    if ((this->actor.bgCheckFlags & BGCHECKFLAG_WALL) && (ABS(targetStepVal) >= 0x4000)) {
+        targetStepVal =
+            (this->actor.yawTowardsPlayer > 0) ? this->actor.wallYaw - 0x4000 : this->actor.wallYaw + 0x4000;
+        Math_SmoothStepToS(&this->actor.world.rot.y, targetStepVal, 1, stepVal, 0);
     } else {
-        Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 1, phi_a3, 0);
+        Math_SmoothStepToS(&this->actor.world.rot.y, this->actor.yawTowardsPlayer, 1, stepVal, 0);
     }
     this->actor.shape.rot.y = this->actor.world.rot.y;
     yawDiff = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
     if ((ABS(yawDiff) <= temp_t0) && (this->actor.xzDistToPlayer < 100.0f)) {
         if (ABS(this->actor.yDistToPlayer) < 150.0f) {
             if ((globalCtx->gameplayFrames & 1)) {
-                func_80A74E2C(this);
+                EnIk_SetupVerticleAttack(this);
             } else {
-                func_80A751C8(this);
+                EnIk_SetupDoubleHorizontalAttack(this);
             }
         }
     }
-    if (func_80A74674(globalCtx, &this->actor) != NULL) {
-        func_80A751C8(this);
-        this->unk_2FC = 1;
+    if (EnIk_CheckForIronObj(globalCtx, &this->actor) != NULL) {
+        EnIk_SetupDoubleHorizontalAttack(this);
+        this->isDestroyingIronObj = 1;
```

Comment:
`isDestroyingIronObj` can use true/false afaict

also on naming the variable, instead of `IronObj`,
I'd either go with something slightly more readable like `IKObject` "iron knuckle object",
or with (`BgJya`)`Ironobj` from the internal file name (`ovl_Bg_Jya_Ironobj`)

Reply:
Yeah, IronObj came from trying to merge both of those while keeping the name short. I'll just go with `isDestroyingBgJyaIronobj` so it's easier to keep track of what it is interacting with specifically.